gpytorch/kernels/grid_kernel.py

```diff
@@ -1,179 +1,176 @@
 #!/usr/bin/env python3
 
 import warnings
-from typing import Optional
+from typing import Iterable, Union
 
 import torch
+from jaxtyping import Float
 from linear_operator import to_dense
-from linear_operator.operators import KroneckerProductLinearOperator, ToeplitzLinearOperator
+from linear_operator.operators import KroneckerProductLinearOperator, LinearOperator, ToeplitzLinearOperator
 from torch import Tensor
 
 from .. import settings
-from ..utils.grid import convert_legacy_grid, create_data_from_grid
+from ..utils.grid import create_data_from_grid
 from .kernel import Kernel
 
 
 class GridKernel(Kernel):
     r"""
-    If the input data :math:`X` are regularly spaced on a grid, then
-    `GridKernel` can dramatically speed up computatations for stationary kernel.
-
-    GridKernel exploits Toeplitz and Kronecker structure within the covariance matrix.
+    `GridKernel` wraps a stationary kernel that is computed on a (multidimensional)
+    grid that is regularly spaced along each dimension.
+    It exploits Toeplitz and Kronecker structure within the covariance matrix
+    for massive computational speedups.
     See `Fast kernel learning for multidimensional pattern extrapolation`_ for more info.
 
     .. note::
 
         `GridKernel` can only wrap **stationary kernels** (such as RBF, Matern,
         Periodic, Spectral Mixture, etc.)
 
-    Args:
-        base_kernel (Kernel):
-            The kernel to speed up with grid methods.
-        grid (Tensor):
-            A g x d tensor where column i consists of the projections of the
-            grid in dimension i.
-        active_dims (tuple of ints, optional):
-            Passed down to the `base_kernel`.
-        interpolation_mode (bool):
-            Used for GridInterpolationKernel where we want the covariance
-            between points in the projections of the grid of each dimension.
-            We do this by treating `grid` as d batches of g x 1 tensors by
-            calling base_kernel(grid, grid) with last_dim_is_batch to get a d x g x g Tensor
-            which we Kronecker product to get a g x g KroneckerProductLinearOperator.
+    :param base_kernel: The stationary kernel to speed up with grid methods.
+    :param grid: A list of tensors where tensor `i` consists of the projections
+        of the grid in dimension i.
+    :param active_dims:
+
+    :ivar ragged_grid: A concatenation of all grid projections
+    :type ragged_grid: Tensor (max(M_i) x D)
+    :ivar full_grid: A full representation of the grid
+    :type ragged_grid: Tensor (N x D)
 
     .. _Fast kernel learning for multidimensional pattern extrapolation:
         http://www.cs.cmu.edu/~andrewgw/manet.pdf
     """
 
+    # TODO: update doc
```

Comment:
Is this leftover from work during the PR? If not could you specify what part of the docs needs to be updated so potentially somebody else can fix this in the future?

```suggestion
```